packages/flutter/lib/src/widgets/safe_area.dart

```diff
@@ -0,0 +1,180 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'basic.dart';
+import 'debug.dart';
+import 'framework.dart';
+import 'implicit_animations.dart';
+import 'media_query.dart';
+
+/// A widget that insets its child by sufficient padding to avoid
+/// intrusions by the operating system.
```

Comment:
In addition to OS-specific intrusions, Is it worth explicitly adding something about physical device screens whose features may intrude upon the view?

Reply:
I've added a brief paragraph about The Notch.